src/reducers/notificationReducer.js

```diff
@@ -0,0 +1,50 @@
+import * as notificationActions from 'actions/notificationActions'
+
+const initialState = {
+  receivePushNotificationsEnabled: true,
+  list: []
+}
+
+export default function notificationReducer (state = initialState, action) {
+  switch (action.type) {
+    case notificationActions.EMIT_NOTIFICATION:
+      return {
+        ...state,
+        list: state.receivePushNotificationsEnabled ? state.list.concat(action.notification) : []
```

Comment:
It can be dangerous for a reducer to rely too much on its own state to determine how it responds to actions. Including this dependency on receivePushNotificationsEnabled means that we can never break this reducer into multiple using combineReducers (such as to have a more complex set of notification options as a separate reducer to the list of notifications). I think it's better to avoid dispatching actions if we don't intend for them to alter the state (or to be used expressly for debugging, like rawHttp and json actions). I'd prefer to have the notification component pull receivePushNotificationsEnabled out of state and have the list reducer separate from the options reducer. This has the added benefit of significantly simplifying testing of the reducer, as there are fewer cases to account for overall.

Reply:
Interesting, let's talk more about this when you have a minute.